tests/test_request_fingerprinter.py

```diff
@@ -0,0 +1,341 @@
+import sys
+from unittest.mock import patch
+
+import pytest
+from packaging.version import Version
+from scrapy import __version__ as SCRAPY_VERSION
+
+if Version(SCRAPY_VERSION) < Version("2.7"):
+    pytest.skip("Skipping tests for Scrapy < 2.7", allow_module_level=True)
+
+from importlib.metadata import version as package_version
+
+from scrapy import Request, Spider
+from scrapy.http import Response
+from web_poet import HttpResponse, ItemPage, PageParams, RequestUrl, WebPage
+
+from scrapy_poet import DummyResponse, ScrapyPoetRequestFingerprinter
+from scrapy_poet._request_fingerprinter import _serialize_dep
+from scrapy_poet.utils.testing import get_crawler as _get_crawler
+
+ANDI_VERSION = Version(package_version("andi"))
+
+SETTINGS = {
+    "DOWNLOADER_MIDDLEWARES": {
+        "scrapy_poet.InjectionMiddleware": 543,
+    },
+    "REQUEST_FINGERPRINTER_CLASS": ScrapyPoetRequestFingerprinter,
+}
+
+
+def get_crawler(spider_cls=None, settings=None):
+    settings = SETTINGS if settings is None else settings
+    kwargs = {}
+    if spider_cls is not None:
+        kwargs["spider_cls"] = spider_cls
+    return _get_crawler(settings=settings, **kwargs)
+
+
+def test_single_callback():
+    class TestSpider(Spider):
+        name = "test_spider"
+
+        async def parse_page(self, response, page: WebPage):
+            pass
+
+    crawler = get_crawler(spider_cls=TestSpider)
+    fingerprinter = crawler.request_fingerprinter
+    request1 = Request("https://toscrape.com")
+    fingerprint1 = fingerprinter.fingerprint(request1)
+    request2 = Request("https://toscrape.com", callback=crawler.spider.parse_page)
+    fingerprint2 = fingerprinter.fingerprint(request2)
+    request3 = Request("https://toscrape.com", callback=crawler.spider.parse_page)
+    fingerprint3 = fingerprinter.fingerprint(request3)
+    request4 = Request("https://books.toscrape.com", callback=crawler.spider.parse_page)
+    fingerprint4 = fingerprinter.fingerprint(request4)
+    assert fingerprint1 != fingerprint2  # same url, no deps vs deps
+    assert fingerprint2 == fingerprint3  # same url, same callback
+    assert fingerprint2 != fingerprint4  # different url, same callback
```

Comment:
I think let's also add a test case for `Request.meta` where it doesn't affect the fps as well.

Although I'm not too sure about this since if the `"page_params"` key is present, it would create `PageParams` which _might_ affect how a Page Object behaves. For example, a boolean flag is passed to run additional requests.

This is a complicated example and might be too specific. But having the fp equality tests for now would be useful later on when we decide to do something about it.